dbcon/joblist/tupleannexstep.cpp

```diff
@@ -1072,11 +1099,25 @@ void TupleAnnexStep::finalizeParallelOrderBy()
     dlTimes.setEndOfInputTime();
     printCalTrace();
   }
+
+  // Release memory before ctor
+  for (uint64_t id = 1; id <= fMaxThreads; id++)
+  {
+    fOrderByList[id]->returnAllRGDataMemory2RM();
+  }
+
+  StepTeleStats sts;
+  sts.query_uuid = fQueryUuid;
+  sts.step_uuid = fStepUuid;
+  sts.msg_type = StepTeleStats::ST_SUMMARY;
```

Comment:
A bit of SOLID should work here

Reply:
I didn't get how SOLID is related to this but I will add a specialized ctor for the purpose as an external patch b/c the change isn't directly related to this patch.

Reply:
Addressed with a separate PR